sectionproperties/pre/sections.py

```diff
@@ -470,11 +477,11 @@ def split_section(
 
         return (top_right_geoms, bottom_left_geoms)
 
-    def offset_section_perimeter(self, amount: float = 0, resolution: float = 12):
+    def offset_perimeter(self, amount: float = 0, resolution: float = 12):
```

Comment:
@connorferster, this method seems to fail if you have a compound geometry you want to offset without any specifically defined material(s)?

Have not checked with materials if it works.

Can you please check and see if you're seeing the same thing?

Reply:
Actually I think it's more to do with the fact that materials have not been implemented in CompoundGeometry yet?

```
AttributeError: 'CompoundGeometry' object has no attribute 'material'
```

Reply:
Hi @connorferster
I'm not so sure this is working correctly or as intended with hollow sections, a negative buffer seems to reduce the interior and exterior, and a positive buffer seems to increase both the interior and exterior?
For example
```
geometry1 = sections.circular_hollow_section(200, 10, 40)
geometry1 = geometry1.offset_perimeter(amount=-4)
geometry1.plot_geometry()
```
The original intent was that it reduced the exterior only to model corrosion of a perimeter?

Perhaps another argument is required to distinguish between wanting to increase or decrease interior/exterior/both so as not to limit it's use (for example may want interior to corrode if you're dealing with a pipeline or something)?

Reply:
Hi @Agent6-6-6,

Thanks for checking this out. I confirm that I am seeing the same thing. That is correct, `.material` is not implemented on CompoundGeometry. The intention is that individual geometries would be assigned materials and then combined to create a compound. 

The problem was that I had changed the name of the method from `.offset_section_perimeter()` to `.offset_perimeter()` but I forgot to change the method name in `CompoundGeometry`, also. Now, the method runs. 

You bring up a good point about the method. Agreed, it would be better if you could specify whether to offset the exterior, the interiors, or both. I am working on implementing this now. Will be coming in next commit.